src/scripts/DeployToken.ts

```diff
@@ -0,0 +1,429 @@
+  image: string,
+  metadata: string,
+  context: string,
+  supply: bigint,
+  salt: `0x${string}`,
+  initialSupplyChainId: bigint
+): Promise<`0x${string}`> {
+  const create2Salt = keccak256(
+    encodeAbiParameters(
+      [{ type: 'address' }, { type: 'bytes32' }],
+      [admin, salt]
+    )
+  );
+
+  const constructorArgs = encodeAbiParameters(
+    [
+      { type: 'string' },
+      { type: 'string' },
+      { type: 'uint256' },
+      { type: 'address' },
+      { type: 'string' },
+      { type: 'string' },
+      { type: 'string' },
+      { type: 'uint256' },
+    ],
+    [
+      name,
+      symbol,
+      supply,
+      admin,
+      image,
+      metadata,
+      context,
+      initialSupplyChainId,
+    ]
+  );
+
+  const artifact = await import('./ClankerToken.json');
+  const creationCode = artifact.bytecode.object as `0x${string}`;
+  const encodedCreationCode = encodePacked(
+    ['bytes', 'bytes'],
+    [creationCode, constructorArgs]
+  );
+  const creationCodeHash = keccak256(encodedCreationCode);
+
+  const hash = keccak256(
+    encodePacked(
+      ['uint8', 'address', 'bytes32', 'bytes32'],
+      [0xff, CLANKER_FACTORY_V3_1, create2Salt, creationCodeHash]
+    )
+  );
+
+  return `0x${hash.slice(-40)}` as `0x${string}`;
+}
+
+export async function generateSalt_v3_1(
+  admin: `0x${string}`,
+  name: string,
+  symbol: string,
+  image: string,
+  metadata: string,
+  context: string,
+  supply: bigint,
+  pairedTokenAddress: `0x${string}`,
+  initialSupplyChainId: bigint
+): Promise<{ salt: `0x${string}`; token: `0x${string}` }> {
+  const startingPoint = BigInt(
+    '0x' +
+      Array.from(crypto.getRandomValues(new Uint8Array(32)))
+        .map((b) => b.toString(16).padStart(2, '0'))
+        .join('')
+  );
+  let i = startingPoint;
+  while (true) {
+    const salt = `0x${i.toString(16).padStart(64, '0')}` as `0x${string}`;
+
+    const token = await predictToken_v3_1(
+      admin,
+      name,
+      symbol,
+      image,
+      metadata,
+      context,
+      supply,
+      salt,
+      initialSupplyChainId
+    );
+
+    const tokenNum = BigInt(token);
+    const pairedTokenNum = BigInt(pairedTokenAddress);
+
+    if (tokenNum < pairedTokenNum) {
+      return { salt, token };
+    }
+
+    i += BigInt(
+      Math.floor(crypto.getRandomValues(new Uint8Array(1))[0] % 1000) + 1
+    );
+  }
+}
+
+// Dummy transaction sender - replace with actual implementation (viem or ethers)
+async function sendTransactionAsync({
+  to,
+  data,
+  value,
+  gas,
+}: {
+  to: `0x${string}`;
+  data: `0x${string}`;
+  value: bigint;
+  gas: bigint;
+}): Promise<string> {
+  console.log('Sending transaction:', { to, data, value, gas });
+  // send transaction
+  return '0xTransactionHash';
+}
+
+/**
+ * Builds a token deployment transaction
+ */
+async function buildDeploymentTransaction({
+  deployerAddress,
+  formData,
+  chainId,
+  userInfo,
+  desiredPrice = 10, // Default desired price (pairs with WETH)
```

Comment:
dumb question, what does this price 'mean'? Like is it one of WETH for 10 of the clanker?